Add NYXT NYX405V1 target

Nyxtronics opened the pull request:
Add new target: NYXT NYX405V1.
Tested with a real hardware:
![IMG_5177](https://github.com/user-attachments/assets/f887529c-1b67-45fe-a63e-785bcaa78928)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added support for the NYX405V1 flight controller board.
  * Enabled onboard IMU, barometer, external flash and OSD support.
  * Exposed multiple UART/I2C/SPI ports, ADC inputs, and PWM/servo outputs.
  * Provided sensible defaults for blackbox logging, ESC telemetry (mapped to a serial port), motor/ESC behavior, meter scaling, beeper inversion, and sensor alignment.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/betaflight/config/pull/681?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Nyxtronics committed:
Create config.h

Initial commit of NYX405V1 flight controller config

Nyxtronics committed:
Merge branch 'betaflight:master' into FC-NYXT-F405-v1

Nyxtronics committed:
Update config.h

Added: USE_MAG,  ADC1_DMA_OPT

nerdCopter commented:
is the 6S version and the 8S version the same target config.h?

Nyxtronics commented:
Yes, targets are the same.
The 6S and 8S boards use different DC-DC converters, their functionality is identical.

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,107 @@
+#define DEFAULT_CURRENT_METER_SCALE 240
+#define DEFAULT_VOLTAGE_METER_SCALE 110
+#define BEEPER_INVERTED
+#define USE_FLASH
```
Duplicate
```suggestion

```

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,107 @@
+#define BARO_I2C_INSTANCE I2CDEV_1
+#define DEFAULT_BARO_I2C_ADDRESS 119
+#define DEFAULT_BLACKBOX_DEVICE BLACKBOX_DEVICE_FLASH
+#define DEFAULT_DSHOT_BURST DSHOT_DMAR_ON
```
F405 should not have burst DMA enabled
```suggestion
#define DEFAULT_DSHOT_BURST DSHOT_DMAR_OFF
```

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,107 @@
+#define USE_MAX7456
+#define USE_BARO
+#define USE_BARO_DPS310
+#define USE_MAG
```
Do not include defines for hardware not included on FC.
```suggestion

```

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,107 @@
+#define MOTOR3_PIN PA3
+#define MOTOR4_PIN PA2
+#define RX_PPM_PIN PA15
+#define SERVO1_PIN PB5
```
Servo conflicts with M1 and M2 on TIM3.
```suggestion

```

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,107 @@
+#define GYRO_1_ALIGN CW90_DEG
+#define DEFAULT_FEATURES (FEATURE_OSD | FEATURE_LED_STRIP | FEATURE_GPS | FEATURE_ESC_SENSOR )
+
+#define SERIALRX_UART SERIAL_PORT_USART1
```
UARTs 1 & 3 are susceptible to DFU hijacking and should not be used for receivers or GPS.
```suggestion

```

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,107 @@
+#define MAX7456_SPI_INSTANCE SPI1
+#define GYRO_1_SPI_INSTANCE SPI3
+#define GYRO_1_ALIGN CW90_DEG
+#define DEFAULT_FEATURES (FEATURE_OSD | FEATURE_LED_STRIP | FEATURE_GPS | FEATURE_ESC_SENSOR )
```
Why are you including these in the config?

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,107 @@
+#define GYRO_1_EXTI_PIN PB8
+#define GYRO_1_CS_PIN PB3
+
+#define TIMER_PIN_MAPPING                                                      \
+  TIMER_PIN_MAP(0, PB1, 1, 0)                                                  \
+  TIMER_PIN_MAP(1, PB0, 1, 0)                                                  \
+  TIMER_PIN_MAP(2, PA2, 1, 0)                                                  \
+  TIMER_PIN_MAP(3, PA3, 1, 1)                                                  \
+  TIMER_PIN_MAP(4, PC8, 2, 0)                                                  \
+  TIMER_PIN_MAP(5, PA15, 1, 0)                                                 \
+  TIMER_PIN_MAP(6, PB5, 1, 0)
```
```suggestion
#define TIMER_PIN_MAPPING                                                      \
  TIMER_PIN_MAP(0, PB1, 2, 0)                                                  \
  TIMER_PIN_MAP(1, PB0, 2, 0)                                                  \
  TIMER_PIN_MAP(2, PA3, 1, 1)                                                  \
  TIMER_PIN_MAP(3, PA2, 1, 0)                                                  \
  TIMER_PIN_MAP(4, PC8, 2, 0)                                                  
```

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,107 @@
+#define MOTOR2_PIN PB0
+#define MOTOR3_PIN PA3
+#define MOTOR4_PIN PA2
+#define RX_PPM_PIN PA15
```
Conflicts with M3 and M4 on TIM2
```suggestion

```

Nyxtronics committed:
Apply suggestions from code review

Changes from code review

Co-authored-by: ot0tot <36753790+ot0tot@users.noreply.github.com>

Nyxtronics committed:
Update config.h

Removed default features as requested

Nyxtronics committed:
Update config.h

Remapped SERVO pin

sugaarK commented:
@ot0tot did this pass the review?

Nyxtronics committed:
Update config.h

Removed GPS serial port define

Nyxtronics committed:
Update config.h

Timer map numbering fix

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,101 @@
+
+#define FC_TARGET_MCU STM32F405
+
+#define BOARD_NAME NYX405V1
+#define MANUFACTURER_ID NYXT
+
```
@coderabbitai This is not correct. Manufacturer ID is 4 characters.

nerdCopter reviewed (approved):
Approving for workflow, deferring to target-team

osirisinferi commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,101 @@
+#define FLASH_CS_PIN PB12
+#define MAX7456_SPI_CS_PIN PA4
+#define GYRO_1_EXTI_PIN PB8
+#define GYRO_1_CS_PIN PB3
```
```suggestion
#define GYRO_1_CS_PIN PA4
```

IMU is on SPI1 now.

osirisinferi commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,101 @@
+#define ADC_RSSI_PIN PC5
+#define ADC_CURR_PIN PC3
+#define FLASH_CS_PIN PB12
+#define MAX7456_SPI_CS_PIN PA4
```
```suggestion
#define MAX7456_SPI_CS_PIN PB3
```
OSD IC was changed to SPI3.

osirisinferi commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,101 @@
+#define MAX7456_SPI_CS_PIN PA4
+#define GYRO_1_EXTI_PIN PB8
+#define GYRO_1_CS_PIN PB3
+#define PINIO1_PIN PA15
```
```suggestion
#define PINIO1_PIN PA15
#define PINIO2_PIN PB4
```
`+9V_EN` was added. You might want to consider swapping these around to make the 9V BEC PINIO the first and the other GPIO the second USER box.

Nyxtronics replied:
Swapped GPIOs as recommended

osirisinferi commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,101 @@
+#define ADC1_DMA_OPT        1
+#define MAG_I2C_INSTANCE I2CDEV_1
+#define BARO_I2C_INSTANCE I2CDEV_1
+#define DEFAULT_BARO_I2C_ADDRESS 119
```
```suggestion
```
SDO has been pulled to GND so the BF default of 0x76 / 118 is appropriate.

osirisinferi commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,101 @@
+#define FLASH_SPI_INSTANCE SPI2
+#define SYSTEM_HSE_MHZ 8
+#define MAX7456_SPI_INSTANCE SPI1
+#define GYRO_1_SPI_INSTANCE SPI3
```
```suggestion
#define MAX7456_SPI_INSTANCE SPI3
#define GYRO_1_SPI_INSTANCE SPI1
```

osirisinferi commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,101 @@
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE 240
+#define DEFAULT_VOLTAGE_METER_SCALE 110
```
```suggestion
```
110 is the default.

osirisinferi commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,101 @@
+#define DEFAULT_DSHOT_BURST DSHOT_DMAR_OFF
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE 240
```
Is this board going to be part of a stack with known current meter scale? I noticed just 1 out of 3 of your ESCs have 240 as scale, the other 2 having 80.

Nyxtronics replied:
Changed to 80 as this is a more commonly used configuration.

Nyxtronics committed:
Changes to the hardware pinout

SPI1 <-> SPI3 (OSD <-> AXL)
Added 9V BEC EN pin
Added USER GPIO pin
Changed default current divider value 240 -> 80
Changed baro default address: 119 -> 118

Nyxtronics committed:
Update config.h

Added USER1 PINIO BOX name as "9V BEC"

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,104 @@
+#define MAG_I2C_INSTANCE I2CDEV_1
+#define BARO_I2C_INSTANCE I2CDEV_1
+#define DEFAULT_BLACKBOX_DEVICE BLACKBOX_DEVICE_FLASH
+#define DEFAULT_DSHOT_BURST DSHOT_DMAR_OFF
```
```suggestion
#define DEFAULT_DSHOT_BURST DSHOT_DMAR_OFF
#define DEFAULT_DSHOT_BITBANG DSHOT_BITBANG_OFF
```

ot0tot commented on configs/NYX405V1/config.h:
```diff
@@ -0,0 +1,104 @@
+#define DEFAULT_DSHOT_BURST DSHOT_DMAR_OFF
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE 80
```
Should be removed unless this is being sold as part of a stack
```suggestion

```

Nyxtronics replied:
It will be part of the stack.

ot0tot commented:
Schematics approved. Will approve config after the above changes are accepted.

Nyxtronics committed:
Update configs/NYX405V1/config.h

Co-authored-by: ot0tot <36753790+ot0tot@users.noreply.github.com>

Nyxtronics committed:
Added approved target key

haslinghuis reviewed (approved):
No `BOX_USER2_NAME` definition ?
